xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -927,49 +921,28 @@ unsigned int CAESinkAUDIOTRACK::AddPackets(uint8_t **data, unsigned int frames,
   }
   unsigned int written_frames = static_cast<unsigned int>(written / m_format.m_frameSize);
   double time_to_add_ms = 1000.0 * (CurrentHostCounter() - startTime) / CurrentHostFrequency();
+  // Get Back in Sync with faked pause bursts
   if (m_passthrough && !m_info.m_wantsIECPassthrough)
   {
-    // AT does not consume in a blocking way - it runs ahead and blocks
-    // exactly once with the last package for some 100 ms
-    double extra_sleep = 0.0;
-    if (time_to_add_ms < m_format.m_streamInfo.GetDuration())
-      extra_sleep = (m_format.m_streamInfo.GetDuration() - time_to_add_ms) / 2;
-
-    // if there is still place, just add it without blocking
-    if (m_delay < (m_audiotrackbuffer_sec - (m_format.m_streamInfo.GetDuration() / 1000.0)))
-      extra_sleep = 0;
-
     if (m_pause_ms > 0.0)
     {
-      extra_sleep = 0;
-      m_pause_ms -= m_format.m_streamInfo.GetDuration();
+      // Idea here is: Slowly correct the wrong buffer so that AE should not realize
+      // but do not underrun while doing so
+      double extra_sleep_ms = m_format.m_streamInfo.GetDuration() / 2.0 - time_to_add_ms;
```

Comment:
As commented in forum this improves a lot DTS-HD MA in RAW. 

Tested 4.0 divider and not improves TrueHD and broke DTS-HD MA again. Tested also 1.5 with no good results.

Seems 2.0 is the best value for DTS-HD MA and anyway devices that supports TrueHD also should be able to use IEC and Kodi packer.